src/lua/init.c

```diff
@@ -795,15 +795,7 @@ luaopen_tarantool(lua_State *L)
 	lua_pushstring(L, tarantool_version());
 	lua_setfield(L, LUA_GLOBALSINDEX, "_TARANTOOL");
 
-	/*
-	 * Get tarantool module.
-	 *
-	 * src/lua/init.lua is already registered as tarantool
-	 * module, so we `require` it here, not create.
-	 */
-	lua_getfield(L, LUA_GLOBALSINDEX, "require");
-	lua_pushstring(L, "tarantool");
-	lua_call(L, 1, 1);
+	luaT_newmodule(L, "tarantool", NULL);
```

Comment:
I have reviewed a different patch on the first iteration. It had no modules loading tweaks. I propose to do it in a separate patch.

> In particularly build.asan flag is going to be used in buffer.lua in scope of #7273.
>
> Part of #7327

As I see, buffer.lua is loaded after init.lua, so I don't see why the order of loading should be changed.

There are other changes around the `tarantool` module and I also see no reason behind it. Why it works for `build.target`, but not for `build.asan`?

Maybe it worth to discuss the patch f2f.

BTW, the issue number looks wrong (7273 vs 7327).

Reply:
Fixed wrong PR reference. Also as disscussed f2f split to two commits.